OpenRA.Game/World.cs

```diff
@@ -549,9 +550,21 @@ public void OnPlayerWinStateChanged(Player player)
 			{
 				pi.Outcome = player.WinState;
 				pi.OutcomeTimestampUtc = DateTime.UtcNow;
+
+				if (pi.Outcome == WinState.Lost)
+					DefeatState |= 1UL << gameInfo.Players.IndexOf(pi);
```

Comment:
`DefeatState` on the client side is used for the sync packet rather than replays, so it doesn't feel right to be basing this off `gameInfo.Players`.

IMO the most robust solution would be to build the defeat state inside `OrderManager.Tick` (enumerate over `World.Players` using a standard for loop and set the bits) and remove `World.DefeatState` instead of tracking a processed duplicate of the win states that might end up disagreeing if the code changes in the future.

Reply:
I expect this change will break things server-side, because both sides currently expect the defeat state to be ordered by the replay metadata order, not the player order. This will be missing any scripted players and generally won't match the order of `World.Players`. I expect this will cause problems in the future if not already, so we will need to fix this now before we can proceed.

I am currently investigating solutions, and will reply once I have further advice.

Reply:
25f87e67fadba4890cf91628ca44d0395421a59e is what I had in mind, but this isn't quite working (the metadata block is no longer saved in server replays) and i've run out of time to spend on this today.